code/datums/riding.dm

```diff
@@ -111,39 +111,39 @@
 		ridden.layer = OBJ_LAYER
 
 /datum/riding/atv/turret
-	var/obj/machinery/porta_turret/syndicate/vehicle_turret/turret = null
+	var/obj/vehicle/atv/turret/R = null
 
 /datum/riding/atv/turret/handle_vehicle_layer()
 	if(ridden.dir == SOUTH)
 		ridden.layer = ABOVE_MOB_LAYER
 	else
 		ridden.layer = OBJ_LAYER
 
-	if(turret)
+	if(R.turret)
```

Comment:
elevate turret to obj/vehicle to avoid this

more than atvs are gonna get turrets at some point anyways.